docs/topics/compatibility-guides/compatibility-guide-20.md

```diff
@@ -0,0 +1,200 @@
+| [KT-58589](https://youtrack.jetbrains.com/issue/KT-58589) | Deprecate missed MUST_BE_INITIALIZED when no primary constructor is presented or when class is local                            |
+| [KT-61182](https://youtrack.jetbrains.com/issue/KT-61182) | Unit conversion is accidentally allowed to be used for expressions on variables + invoke resolution                            |
+| [KT-62998](https://youtrack.jetbrains.com/issue/KT-62998) | Forbid assignment of a nullable to a not-null Java field as a selector of unsafe assignment                                    |
+| [KT-57600](https://youtrack.jetbrains.com/issue/KT-57600) | Forbid overriding of Java method with raw-typed parameter with generic typed parameter                                                                                                                               |
+| [KT-47313](https://youtrack.jetbrains.com/issue/KT-47313) | Change (V)::foo reference resolution when V has a companion                                                       |
+| [KT-54997](https://youtrack.jetbrains.com/issue/KT-54997) | Forbid implicit non-public-API accesses from public-API inline function                                                       |
+| [KT-57422](https://youtrack.jetbrains.com/issue/KT-57422) | K2: Prohibit use-site 'get' targeted annotations on property getters                                                       |
+| [KT-34372](https://youtrack.jetbrains.com/issue/KT-34372) | Report missed error for virtual inline method in enum classes                                                       |
+| [KT-47986](https://youtrack.jetbrains.com/issue/KT-47986) | Forbid implicit inferring a type variable into an upper bound in the builder inference context                                                       |
+| [KT-53982](https://youtrack.jetbrains.com/issue/KT-53982) | Keep nullability when approximating local types in public signatures                                                       |
+
+## Tools
+
+### Visibility changes in Gradle
+
+> **Issue**: [KT-67227](https://youtrack.jetbrains.com/issue/KT-67227)
+>
+> **Component**: Gradle
+>
+> **Incompatible change type**: source
+>
+> **Short summary**: Previously, certain Kotlin DSL functions and properties intended for a specific DSL context would 
+> inadvertently leak into other DSL contexts. We've added the `@KotlinGradlePluginDsl` annotation,
+> which prevents the exposure of the Kotlin Gradle plugin DSL functions and properties to levels where they are not intended
+> to be available. The following levels are separated from each other:
+> * Kotlin extension
+> * Kotlin target
+> * Kotlin compilation
+> * Kotlin compilation task
+>
+> **Deprecation cycle**:
+>
+> - 2.0.0: if a build script is configured incorrectly, report a warning
```

Comment:
actually error initially. For most popular use-cases we introduced deprecated methods in the contexts where they could be misused, but it is too much work to do for every case.  And, unfortunately, it is not possible to control the level of violation.